transcom/mymove
# INT B-21073 part dos

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21073)

## Summary

[First INT PR](https://github.com/transcom/mymove/pull/13853)

This is adding a little cleanup + one missed oopsie of validating the country when the Prime attempts to create an address that contains a country code that is not supported.

Since we do not support international moves at this time, I added in a check that IF the Prime provides a country code that is not `US`, then we are going to send back an error message saying the country code they provided is not supported.

### How to test

1. Access MM as Prime Sim
2. Find a move that you can create a shipment for
3. Endpoint would be `/prime/v3/mto-shipments`
4. Try to pass in an address that has a `Country` value of something that is NOT `US` - should get back a validation error
```
{
  "moveTaskOrderID": "0f21be2a-5045-4e90-8212-95a92de90056",
  "requestedPickupDate": "2024-10-12",
  "primeEstimatedWeight": 4000,
  "customerRemarks": "handle with care",
  "pickupAddress": {
    "streetAddress1": "123 OtherWorldly Ave",
    "city": "Not a US Country",
    "state": "AK",
    "postalCode": "90210",
    "country": "GB"
  },
  "destinationAddress": {
    "streetAddress1": "123 Main Ave",
    "streetAddress2": "Apartment 9000",
    "streetAddress3": "Montmârtre",
    "city": "Anytown",
    "state": "AL",
    "postalCode": "90210",
    "country": "US"
  },
  "shipmentType": "HHG"
}
```

![Screenshot 2024-10-11 at 3 55 00 PM](https://github.com/user-attachments/assets/4407105d-b968-4584-8291-1dba571e536d)
